src/main/java/org/jabref/logic/exporter/AtomicFileOutputStream.java

```diff
@@ -59,24 +60,45 @@ public class AtomicFileOutputStream extends FilterOutputStream {
      * The file to which writes are redirected to.
      */
     private final Path temporaryFile;
+
     private final FileLock temporaryFileLock;
     /**
      * A backup of the target file (if it exists), created when the stream is closed
      */
     private final Path backupFile;
+
     private final boolean keepBackup;
 
+    private boolean errorDuringWrite = false;
+
     /**
      * Creates a new output stream to write to or replace the file at the specified path.
      *
      * @param path       the path of the file to write to or replace
      * @param keepBackup whether to keep the backup file after a successful write process
      */
     public AtomicFileOutputStream(Path path, boolean keepBackup) throws IOException {
-        super(Files.newOutputStream(getPathOfTemporaryFile(path)));
+        // Files.newOutputStream(getPathOfTemporaryFile(path)) leads to a "sun.nio.ch.ChannelOutputStream", which does not offer "lock"
```

Comment:
Useless comment

Reply:
No, the comment is indeed useful because it makes clear why we chose the old io and not the nio one